dialog/fileicon.go

```diff
@@ -0,0 +1,127 @@
+package dialog
```

Comment:
nit: please order this file like #827 to make it easier to read

Reply:
I'm going to up this from "nit" - the code may not all be ordered correctly, but new files probably should be or we have an increasing tidyup later.